### PR TITLE
feat(table): table component

### PR DESCRIPTION
**Pull request by TimKolberger**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added
- [x] Docs have been reviewed and added
- [ ] Guide in the docs for usage with a headless table library like react-table

## Pull request type

Adds a themeable visual-only table component. 

Please check the type of change your PR introduces:

- [x] Feature

## Does this introduce a breaking change?

- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -143,6 +144,7 @@
     "react-lorem-component": "^0.13.0",
     "react-router-dom": "^5.2.0",
     "react-spinners": "^0.9.0",
+    "react-table": "^7.6.2",
```
Comment (TimKolberger) [inline package.json:147]: This dependency is just for usage in storybook

**Comment by with-heart** [comment]
Feedback from team meeting:

- Allow `Table` to expand to fit width by default
- Keep explicit `isNumeric` prop for right-aligned `Td`

**Comment by TimKolberger** [comment]
@chakra-ui/react-team  What do think of merging this PR to allow contributors adding features to it?

I will add a guide how to use this component with `react-table` as additional PR.

**Comment by with-heart** [comment]
If you rebase, the failing test should be resolved

**Review by with-heart — APPROVED** [review]

**Pull request merged**